src/main/java/stanhebben/zenscript/type/natives/JavaMethod.java

```diff
@@ -175,6 +179,65 @@ public static IJavaMethod select(boolean doStatic, List<IJavaMethod> methods, IE
         return result;
     }
     
+    /**
+     * Creates the value for the optional Expression
+     *
+     * @param position    position
+     * @param javaMethod  Method to calculate for
+     * @param parameterNr which parameter
+     * @param environment ZS environment (used for ZS types and Expression creation)
+     *
+     * @return default optional (0, false, null) or default value according to the @Optional annotation
+     */
+    private static Expression getOptionalValue(ZenPosition position, JavaMethod javaMethod, int parameterNr, IEnvironmentGlobal environment) {
+        Parameter parameter = javaMethod.getMethod().getParameters()[parameterNr];
+        Optional optional = parameter.getAnnotation(Optional.class);
+        
+        //No optional or empty value --> default value
+        if(optional == null || optional.value().isEmpty()) {
+            return javaMethod.getParameterTypes()[parameterNr].defaultValue(position);
+        }
+        
+        //Method Class given --> Find methodClass method with name methodName and String parameter
+        if(optional.methodClass() != Optional.class) {
+            Class<?> parameterType = parameter.getType();
+            Class<?> valueClass = optional.methodClass();
+            try {
+                Method method = valueClass.getMethod(optional.methodName(), String.class);
+                if(!parameterType.isAssignableFrom(method.getReturnType())) {
+                    environment.error(position, "Optional Annotation Error, cannot assign " + parameterType + " from " + method);
+                    return new ExpressionInvalid(position);
+                }
+                return new ExpressionCallStatic(position, environment, new JavaMethod(method, environment.getEnvironment().getTypeRegistry()), new ExpressionString(position, optional.value()));
+            } catch(NoSuchMethodException ignored) {
+                //Method not found --> Null
+                return new ExpressionNull(position);
+            }
+            //No method class given --> Either primitive or String. If not --> ExpressionInvalid
+            //If null is wanted, no value should be given to the annotation
+        } else {
+            if(parameter.getType().isPrimitive()) {
+                Class<?> clazz = parameter.getType();
+                if(clazz == int.class || clazz == short.class || clazz == long.class || clazz == byte.class)
+                    return new ExpressionInt(position, Long.parseLong(optional.value()), environment.getType(clazz));
```

Comment:
Instead of parsing from the String to a long you could also add a additional parameter named something like 
`long valueLong() default 0;` and just use that as a default for numbers. (same for double and boolean)
This is to avoid typos, if you write `1,232` instead of `1.232` where your IDE won't find the error but the parser will just error. 
Could also _slightly_ improve performance not having to first parse it from a string, but this effect will be minor